vllm/engine/multiprocessing/client.py

```diff
@@ -399,17 +430,51 @@ def generate(
             prompt_adapter_request: Prompt Adapter request to use
                                             for generation, if any.
         """
-        return self._process_request(inputs, sampling_params, request_id,
+        if inputs is not None:
+            prompt = inputs
+        assert (prompt is not None and sampling_params is not None
+                and request_id is not None)
+
+        return self._process_request(prompt, sampling_params, request_id,
                                      lora_request, trace_headers,
                                      prompt_adapter_request)
 
+    @overload  # DEPRECATED
     def encode(
         self,
-        inputs: PromptInputs,
+        *,
+        inputs: PromptType,
         pooling_params: PoolingParams,
         request_id: str,
         lora_request: Optional[LoRARequest] = None,
         trace_headers: Optional[Mapping[str, str]] = None,
+    ) -> AsyncGenerator[EmbeddingRequestOutput, None]:
+        ...
+
+    @overload
+    def encode(
+        self,
+        prompt: PromptType,
+        pooling_params: PoolingParams,
+        request_id: str,
+        lora_request: Optional[LoRARequest] = None,
+        trace_headers: Optional[Mapping[str, str]] = None,
+    ) -> AsyncGenerator[EmbeddingRequestOutput, None]:
+        ...
```

Comment:
Is this overload really needed (and equivalent for the other methods)? Is it just the fact that `prompt` arg is Optional?

That could just be a comment instead? I.e. that the arg won't be optional in future..

Reply:
They are to clearly specify the two versions of the method. `prompt` arg isn't actually optional, it is only like that in the implementation in order to be compatible with both overloads.